stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -1685,6 +1685,7 @@ function(add_swift_target_library name)
     set(SWIFTLIB_TARGET_SDKS ${SWIFT_SDKS})
   endif()
   list_replace(SWIFTLIB_TARGET_SDKS ALL_APPLE_PLATFORMS "${SWIFT_DARWIN_PLATFORMS}")
+  list(REMOVE_DUPLICATES SWIFTLIB_TARGET_SDKS)
```

Comment:
Why is this part of this change?  Is this because adding the `FREESTANDING` now results in duplicates?

Reply:
Yes --  in particular `FREESTANDING` could be present in `SWIFT_DARWIN_PLATFORMS` and as an explicit target when building the Darwin overlay (as a result of https://github.com/apple/swift/blob/78809726080ce61a4f73c013d542c91d5da35f2e/stdlib/public/Platform/CMakeLists.txt#L21)

I could have added the deduplication in the file above, but this seemed something that could happen in other instances as well.

Reply:
This needs a nice big comment here.

Reply:
@edymtt is it possible to instead of adding in this deduplication, add in an assert that we never have duplicates and change the other code?

Reply:
For now I addressed the issue in Platform/CMakeLists.txt (dbbc99a)

Adding the assert seems to require some changes to `precondition_binary_op` which may require further discussion a bit outside of this PR -- if time allows I will tackled that in another PR.